### PR TITLE
Fixed Twitch streams by using the Helix API

### PR DESCRIPTION
**Pull request by tdeeb**
This PR fixes reading Twitch streams by using the new Helix API and obtaining data from the response.

If the stream is offline, it will return null.

### CODE REVIEW

**Comment by davipatury** [comment]
When is this going to be merged?

**Comment by kantenkugel** [comment]
This won't simply work tho, right? Seeing as helix requires a client id or oauth token for auth/ratelimiting

**Comment by tdeeb** [comment]
@kantenkugel LavaPlayer uses its own Client ID, which can be found at the top of `TwitchStreamAudioSourceManager`.

**Comment by kantenkugel** [comment]
oh... you are right... didn't notice that until now.

I'm not so sure if its smart for LP to statically provide that tho. Seeing as that client id might hit the ratelimit rather quickly if every single LP user uses that client id when doing twitch api calls...
Unless of cause thats a special token with more privs

**Comment by kantenkugel** [comment]
That indeed sounds good. Maybe also some kind of alteration/alternative to `AudioSourceManagers.registerRemoteSources(...)`?

**Comment by tdeeb** [comment]
By using `registerRemoteSources`, I feel it's fitting to use the default client id, as it's simply a helper method. To customize which sources are registered or how they're registered, I believe it's on the user to manually perform what they need.

**Comment by tdeeb** [comment]
I kept `CLIENT_ID` as is in the latest commit for compatibility since it can be accessed from anywhere. The other changes discussed have been added.

File: main/src/main/java/com/sedmelluq/discord/lavaplayer/source/twitch/TwitchStreamAudioSourceManager.java [hunk 1]
```diff
@@ -47,6 +48,16 @@ public TwitchStreamAudioSourceManager() {
...
 
+  /**
+   * Create an instance.
+   * @param clientId The Twitch client id for your application.
+   */
+  public TwitchStreamAudioSourceManager(String clientId)
+  {
+      this();
+      twitchClientId = clientId;
+  }
```
Comment (kantenkugel) [inline main/src/main/java/com/sedmelluq/discord/lavaplayer/source/twitch/TwitchStreamAudioSourceManager.java]: i feel like having a no-arg constructor that *only* calls the arg constructor might be a cleaner solution, which can be done by just having `TwitchStreamAudioSourceManager()` -> `this(CLIENT_ID)` and having the one with client-id set both the `httpInterfaceManager` and the used client id.

File: main/src/main/java/com/sedmelluq/discord/lavaplayer/source/twitch/TwitchStreamAudioSourceManager.java [hunk 2]
```diff
@@ -37,6 +37,7 @@
   private static final Pattern streamNameRegex = Pattern.compile(STREAM_NAME_REGEX);
 
   public static final String CLIENT_ID = "jzkbprff40iqj646a697cyrvl0zt2m6";
```
Comment (kantenkugel) [inline main/src/main/java/com/sedmelluq/discord/lavaplayer/source/twitch/TwitchStreamAudioSourceManager.java]: Renaming this field to something like `DEFAULT_CLIENT_ID` might describe its new functionality more accurate

File: main/src/main/java/com/sedmelluq/discord/lavaplayer/source/twitch/TwitchStreamAudioSourceManager.java (hunk 1 above)
Comment (tdeeb) [reply main/src/main/java/com/sedmelluq/discord/lavaplayer/source/twitch/TwitchStreamAudioSourceManager.java]: My apologies; I'm having issues getting this to compile in IntelliJ. The suggested changes make a lot of sense, and I added them in as well; thank you!

**Comment by kantenkugel** [comment]
LGTM now... Have you tested if everything works?

**Comment by sedmelluq** [comment]
Looks fine to me.

**Pull request merged**